quantum/impl/quantum_task_queue_impl.h

```diff
@@ -172,10 +172,15 @@ TaskQueue::ProcessTaskResult TaskQueue::processTask()
         {
             return ProcessTaskResult(workItem._isBlocked, workItem._blockedQueueRound);
         }
-        
+
+        // set the current task for local-storage queries
+        setCurrentTask(task.get());
```

Comment:
I would use an RAII mechanism here to avoid having to set it to null. This would also cover the case of exception handling.